workflows_observability/activities.py

```diff
@@ -0,0 +1,89 @@
+from atlan_helpers import get_atlan_client
+from application_sdk.activities import ActivitiesInterface
+from application_sdk.observability.logger_adaptor import get_logger
+from temporalio import activity
+from pyatlan.model.enums import AtlanWorkflowPhase
+import os
+from datetime import datetime
+
+logger = get_logger(__name__)
+activity.logger = logger
+
+
+class WorkflowsObservabilityActivities(ActivitiesInterface):
+    @activity.defn
+    async def fetch_workflows_run(self, args: tuple[str, str]) -> dict:
+        """
+        Fetch workflow runs from Atlan within a time range, based on a selected date and output type.
+        If output_type is 'Local', the workflow run results are stored as JSON files under /tmp/workflows.
+
+        Args:
+            args (tuple[str, str]): A tuple containing:
+                - selected_date (str): A date string in 'YYYY-MM-DD' format to calculate the time range.
+                - output_type (str): Type of output, e.g. 'Local'.
+
+        Returns:
+            dict: Not used, but required by Temporal activities API.
+
+        Raises:
+            Exception: If fetching or processing workflows fails.
+        """
+        try:
+            selected_date, output_type = args
+
+            if output_type == "Local":
+                local_directory = "/tmp/workflows"
+                os.makedirs(local_directory, exist_ok=True)
+
+            logger.info(f"Fetching Atlan workflows since: {selected_date}")
+
+            input_date = datetime.strptime(selected_date, "%Y-%m-%d")
+            current_datetime = datetime.now()
+            time_difference = current_datetime - input_date
+            difference_in_hours = int(time_difference.total_seconds() / 3600)
+
+            client = get_atlan_client()
+
+            results = client.workflow.find_runs_by_status_and_time_range(
+                status=[AtlanWorkflowPhase.SUCCESS, AtlanWorkflowPhase.FAILED],
+                started_at=f"now-{difference_in_hours}h",
+            )
+
+            for result in results:
+                if output_type == "Local":
+                    save_result_locally(result, local_directory)
+
+        except Exception as e:
+            logger.error(f"Failed to process workflows: {str(e)}", exc_info=e)
+            raise
+
+
+def save_result_locally(result, local_directory: str) -> None:
```

Comment:
(nit)[optional] - you could rename the `atlan_helpers.py` and move this method into the file. conventionally, we'd want to keep just activity definitions in this file. let me know what you think.